tests/phpunit/integration/DataAccountingConfigTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace DataAccounting\Tests;
+
+use DataAccounting\Config\DataAccountingConfig;
+use MediaWikiIntegrationTestCase;
+
+/**
+ * @group Database
+ */
+class DataAccountingConfigTest extends MediaWikiIntegrationTestCase {
+	private array $configData;
+	private array $configDataOverride;
+
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->configData = [
+			'TestString' => 'test',
+			'TestInt' => 314,
+			'TestArray' => [
+				'test',
+				314
+			],
+			'TestMultiArray' => [
+				'entry1' => 'test',
+				'entry2' => 314
+			]
+		];
+		$this->configDataOverride = [
+			'TestString' => 'test2',
+			'TestInt' => 314159,
+			'TestArray' => [
+				'test2',
+				314159
+			],
+			'TestMultiArray' => [
+				'entry1' => 'test2',
+				'entry2' => 314159
+			]
+		];
+	}
+
+	/**
+	 * @covers \DataAccounting\Config\Handler::configFactoryCallback
+	 */
+	public function testDAConfigFactory() {
+		$config = $this->getServiceContainer()->getConfigFactory()->makeConfig( 'da' );
+		$this->assertInstanceOf( DataAccountingConfig::class, $config );
+	}
+
+	/**
+	 * @covers \DataAccounting\Config\Handler::getConfig
+	 */
+	public function testGetDAConfig() {
+		$handler = $this->getServiceContainer()->getService( 'DataAccountingConfigHandler' );
+		$this->assertInstanceOf( DataAccountingConfig::class, $handler->getConfig() );
+	}
+
+	/**
+	 * @covers \DataAccounting\Config\DataAccountingConfig::get
+	 */
+	public function testExtendMWConfig() {
+		$config = $this->getServiceContainer()->getMainConfig();
+		$daConfig = $this->getServiceContainer()->getConfigFactory()->makeConfig( 'da' );
+		foreach ( $this->configData as $key => $val ) {
+			$this->setMwGlobals( "wg$key", $val );
```

Comment:
Isn't this supposed to be "da$key"?

Reply:
"wg$key" is correct here, as this case tests that keys, that are in the mainconfig are available in the da config as well

Reply:
Should "da$key" namespace be tested as well?

Reply:
oh, do you mean setting globals with prefix "da" should be tested to appear in the config then?
sure, i will add that

Reply:
Yeah exactly.

Reply:
done. i also added a testcase for the config handlers set method